pennylane/plugins/default_qubit.py

```diff
@@ -112,6 +114,9 @@ def apply(self, operations, rotations=None, **kwargs):
                 basis_state = par[0]
                 self.apply_basis_state(basis_state, wires)
 
+            elif isinstance(operation, MultiRZ):
+                self._state = self.vec_vec_product(operation.eigvals, self._state, wires)
```

Comment:
We could expand this if statement to also apply to PauliZ, CZ

Reply:
(+ other diagonal gates)

Reply:
Yes, totally! And we have to do it. The performance increase is just crazy

Reply:
I'd say we have a mixin `DiagonalGate` that exposes `eigvals` (`diagonal` would probably better)